staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/admission_test.go

```diff
@@ -276,7 +279,67 @@ func setupTestCommon(t *testing.T, compiler ValidatorCompiler) (plugin admission
 		// wait for informer factory to shutdown
 		fakeInformerFactory.Shutdown()
 	})
-	return handler, dynamicClient.Tracker(), fakeClient.Tracker(), handler.evaluator.(*celAdmissionController)
+
+	// Wait for admission controller to begin its object watches
+	// This is because there is a very rare (0.05% on my machine) race doing the
+	// initial List+Watch if an object is added after the list, but before the
+	// watch it could be missed.
+	//
+	// This is only due to the fact that NewSimpleClientset above ignores
+	// LastSyncResourceVersion on watch calls, so do it does not provide "catch up"
+	// which may have been added since the call to list.
+	if !cache.WaitForNamedCacheSync("initial sync", testContext.Done(), handler.evaluator.HasSynced) {
+		t.Fatal("failed to do perform initial cache sync")
+	}
+
+	// WaitForCacheSync only tells us the list was performed.
+	// Keep changing an object until it is observable, then remove it
+
+	i := 0
+
+	dummyPolicy := &v1alpha1.ValidatingAdmissionPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "dummypolicy.example.com",
+			ResourceVersion: fmt.Sprint(i),
```

Comment:
Is this line necessary? Seems weird to have to specify a resource version on an object you are about to create.

Reply:
this is just for the test object tracker which has no special handling of resource version. I am only changing it to trigger on update to be noticed by the watcher. Ill change it to an annotation so it doesn't raise eyebrows

Reply:
updated to remove usage of RV